### PR TITLE
Changes to support running E2E tests on minikube.

### PR DESCRIPTION
**Pull request by jlewi**
* Parameterize the E2E workflow with a parameter that controls the platform
   we are running on.

* Related to kubeflow/testing#6

* Add environment variable to configure KUBECONFIG.

* Split test_deploy into separate functions to deploy the kubeflow APP
      and get GKE credentials

* This allows us to handle credentials and the kubeconfig file consistently
      for GKE and minikube; on subsequent steps we just set KUBECONFIG to
      point to the config.

* Refactor the workflow to be a DAG.

* Update the TFServing workflow to use the new version of test_deploy
      which uses a separate step to get credentials and store them in .kube_config.

* The name for the test_runner step needs to be unique so that we will
      create different test cases in gubernator for the different platforms.

* Parameterize the E2E workflow by a platform parameter that controls whether we use minikube or gke.

* Always store kube config on an NFS share.

* Fix lint and add a style .yapf file to autoformat files copied for tf-operator.

Related to kubeflow/testing#6

Depends on kubeflow/testing#83

---
This change is [Reviewable](https://reviewable.io/reviews/kubeflow/kubeflow/523)

### CODE REVIEW

**Comment by jlewi** [comment]
@lluunn Can you review when you have a chance please?

**Comment by lluunn** [comment]
The test failed. It shows 8 successful tests, but I don't see the serving tests.

File: testing/test_deploy.py [hunk 1]
```diff
@@ -367,100 +422,104 @@ def deploy_minikube(args):
...
   # Locate the install minikube script.
-  install_script = os.path.join(os.path.dirname(__file__), "install_minikube.sh")
-  
+  install_script = os.path.join(
+    os.path.dirname(__file__), "install_minikube.sh")
+
   if not os.path.exists(install_script):
-    logging.error("Could not find minikube install script: %s", install_script)
-    
+    logging.error("C %s", install_script)
```
Comment (lluunn) [inline testing/test_deploy.py]: I guess modified by accident?

File: testing/test_deploy.py [hunk 2]
```diff
@@ -367,100 +422,104 @@ def deploy_minikube(args):
...
+
   vm_util.wait_for_vm(args.project, args.zone, args.vm_name)
   vm_util.execute_script(args.project, args.zone, args.vm_name, install_script)
-  vm_util.execute(args.project, args.zone, args.vm_name, ["sudo minikube start --vm-driver=none --disk-size=40g"])
-    
-  # Copy the .kube and .minikube files to test_dir  
+
+  # Copy the .kube and .minikube files to test_dir
+  for target in ["~/.kube"]:
```
Comment (lluunn) [inline testing/test_deploy.py]: for loop for one item

File: testing/test_deploy.py [hunk 3]
```diff
@@ -367,100 +422,104 @@ def deploy_minikube(args):
...
+    os.path.dirname(__file__), "install_minikube.sh")
+
   if not os.path.exists(install_script):
-    logging.error("Could not find minikube install script: %s", install_script)
-    
+    logging.error("C %s", install_script)
+
   vm_util.wait_for_vm(args.project, args.zone, args.vm_name)
   vm_util.execute_script(args.project, args.zone, args.vm_name, install_script)
-  vm_util.execute(args.project, args.zone, args.vm_name, ["sudo minikube start --vm-driver=none --disk-size=40g"])
```
Comment (lluunn) [inline testing/test_deploy.py]: we don't need to start it?

File: .style.yapf [hunk 4]
```diff
@@ -0,0 +1,4 @@
+[style]
+based_on_style = google
+indent_width: 2
+continuation_indent_width: 2
```
[the remark this answers is not in the document]
Comment (jlewi) [reply .style.yapf]: Good catch.

File: testing/test_deploy.py [hunk 5]
```diff
@@ -120,16 +105,22 @@ def setup_kubeflow_ks_app(args, api_client):
...
-    logging.warn("GITHUB_TOKEN not set; you will probably hit Github API "
-                 "limits.")
+    logging.warning("GITHUB_TOKEN not set; you will probably hit Github API "
+                    "limits.")
   # Initialize a ksonnet app.
   app_name = "kubeflow-test-" + uuid.uuid4().hex[0:4]
-  util.run(["ks", "init", app_name,], cwd=args.test_dir)
+  util.run(
+    [
+      "ks",
```
[the remark this answers is not in the document]
Comment (jlewi) [reply testing/test_deploy.py]: This is the autoformatted result. If we don't like it lets open up an issue to try to figure out the proper YAPF style.

File: testing/test_deploy.py [hunk 6]
```diff
@@ -154,51 +145,109 @@ def setup_kubeflow_ks_app(args, api_client):
...
+  util.configure_kubectl(project, zone, cluster_name)
+
+  # TODO(jlewi): If GOOGLE_APPLICATION_CREDENTIALS is set then I think
+  # we want to modify the KUBECONFIG file to remove the GCP commands.
+  # This will allow it to be truly headless and not require gcloud. 
+  # More importantly, kubectl will properly attach auth.info scope so
+  # that RBAC rules can be applied to the email and not the id.
+  # See https://github.com/kubernetes/kubernetes/pull/58141
+  
+  # TODO(jlewi): Make this a flag.
```
[the remark this answers is not in the document]
Comment (jlewi) [reply testing/test_deploy.py]: Done.

File: testing/test_deploy.py [hunk 7]
```diff
@@ -154,51 +145,109 @@ def setup_kubeflow_ks_app(args, api_client):
...
+  cluster_name = args.cluster
+  zone = args.zone
+  logging.info("Using cluster: %s in project: %s in zone: %s", cluster_name,
+               project, zone)
+  # Print out config to help debug issues with accounts and
+  # credentials.
+  util.run(["gcloud", "config", "list"])
+  util.configure_kubectl(project, zone, cluster_name)
+
+  # TODO(jlewi): If GOOGLE_APPLICATION_CREDENTIALS is set then I think
```
[the remark this answers is not in the document]
Comment (jlewi) [reply testing/test_deploy.py]: Done.

File: testing/test_deploy.py (hunk 1 above)
Comment (jlewi) [reply testing/test_deploy.py]: Yes.

File: testing/test_deploy.py (hunk 3 above)
Comment (jlewi) [reply testing/test_deploy.py]: The command happens inside install_script. It needs to happen before the commands to change the permissions.

**Comment by jlewi** [comment]
@lluunn Thanks. Tests are passing. PTAL.

**Pull request merged**